validator/client/validator_propose.go

```diff
@@ -51,6 +52,19 @@ func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pubKey [48]by
 		return
 	}
 
+	history, err := v.db.ProposalHistory(pubKey[:])
+	if err != nil {
+		log.WithError(err).Error("Failed to get proposal history")
+		return
+	}
+
+	if db.HasProposedForEpoch(history, epoch) {
+		log.Info("Tried to sign a double proposal, rejected")
+		return
+	}
+
+	log.WithField("epoch", epoch).Info("Validator has not proposed for this epoch")
```

Comment:
Is this needed? we need to keep the info logs that happen at every epoch very minimal

Reply:
+1 , this isnt necessary